src/core/classes/drop-erc1155-history.ts

```diff
@@ -0,0 +1,36 @@
+import { DropERC1155 } from "@thirdweb-dev/contracts";
+import { ContractWrapper } from "./contract-wrapper";
+import { BigNumberish } from "ethers";
+
+/**
+ * Manages claim conditions for Edition Drop contracts
+ * @public
+ */
+export class DropErc1155History {
+  private contractWrapper;
+
+  constructor(contractWrapper: ContractWrapper<DropERC1155>) {
+    this.contractWrapper = contractWrapper;
+  }
+
+  /** ***************************************
+   * READ FUNCTIONS
+   *****************************************/
+
+  /**
+   * Pulls the list of all addresses that have claimed a particular token
+   *
+   * @beta - This can be very slow for large numbers of token holders
+   *
+   * @param tokenId - The token id to get the claimers of
+   * @returns - A unique list of addresses that claimed the token
+   */
+  public async getAllClaimerAddresses(
+    tokenId: BigNumberish,
+  ): Promise<string[]> {
+    const a = await this.contractWrapper.readContract.queryFilter(
+      this.contractWrapper.readContract.filters.TokensClaimed(null, tokenId),
```

Comment:
btw the reason why i removed these is because this will get limited to a few hundred/thousand blocks depending on the RPC you're using. The hope was to replace this with the graph, but doesn't look like it's happening any time soon right @jakeloo ?

Reply:
jake mentioned that's still gonna be the case so we're just adding these two functions to unblock buildspace and then we can start thinking how to do this with the graph going forward

Reply:
subgraph is probably a couple weeks away. we should unblock buildspace first, then we'll swap out the implementation later.